sinonjs/sinon
# Upgrade @sinonjs/eslint-config

This PR upgrades ~`eslint-config-sinon`~ `@sinonjs/eslint-config` and fixes all uncovered lint violations.

#### Purpose

Adopt the latest `@sinonjs/eslint-config` in order to allow modern JavaScript syntax and ease the maintenance burden of keeping all the development tools up to date.

#### Background

https://hackmd.io/@mroderick/sinon-rfc-drop-legacy-browsers


#### Solution

* Add latest version of `@sinonjs/eslint-config` as `devDependency`
* Fix all lint violations
* Configure pre-commit hook and CI to ignore lint warnings. This allows contributions before we add all the missing JSDoc comments. We can still see all warnings by running `npm run lint`
* Uninstall no longer needed linting dependencies

#### TO DO

- [x] Install `@sinonjs/eslint-config`
- [x] Remove `eslint`
- [x] Remove `eslint-config-prettier`
- [x] Remove `eslint-plugin-ie11`
- [x] Remove `eslint-plugin-mocha`
- [x] Remove `eslint-plugin-prettier`

#### How to verify

1. Check out this branch
2. `npm install`
3. `npm run lint`
4. Observe that the only warnings are `jsdoc/*`
5. Try creating an `async` function without an `await` statement
6. Observe that the linter complains about it
7. Add an `await` statement
8. Observe that the linter allows `async` functions

#### Checklist for author
